dts/arm/infineon/xmc4xxx.dtsi

```diff
@@ -33,8 +33,15 @@
 		compatible = "mmio-sram";
 	};
 
-	flash0: serial-flash@c000000 {
-		compatible = "serial-flash";
+	flash_controller: flash_controller@c000000 {
+		compatible = "infineon,xmc4xxx-flash-controller";
+		erase-value = <0>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		flash0: flash@c000000 {
```

Comment:
This looks incorrect to me. The flash controller and the flash memory itself have the exact same base address? That can't be right, can it? Line 41 here is saying that the first addressable byte of flash storage is at address 0xc000000. But line 36 above is saying that the base address of the register block where you can control the flash itself (like write registers to erase pages, etc.) is the same.

Reply:
yes, good point. I updated the flash controller base address.